src/transformers/training_args.py

```diff
@@ -1545,6 +1549,34 @@ def __post_init__(self):
             if version.parse(version.parse(torch.__version__).base_version) == version.parse("2.0.0") and self.fp16:
                 raise ValueError("--optim adamw_torch_fused with --fp16 requires PyTorch>2.0")
 
+        if is_accelerate_available():
+            if not self.framework == "pt" and not is_torch_available():
```

Comment:
I don't think we need to check if torch is available here for two reasons: 

1) it'd be confusing for a user to set `framework="pt"` and have this error show. Checking the framework vs. env compatibility should be the responsibility of another bit of the code 
2) This won't trigger if someone sets `framework="tf"` and they have torch installed in their env 

```suggestion
            if self.framework != "pt" :
```